File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    let count = {};

    // 개수 세기
    tangerine.forEach(guul => count[guul] ? count[guul]++ : count[guul] = 1);

    // 개수를 기준으로 내림차순 정렬
    const sortedCounts = Object.entries(count).sort((a, b) => b[1] - a[1]);

    let total = 0;
    for (let [, value] of sortedCounts) {
        if (total >= k) break;
        total += value;
        answer++;
    }

    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter
def solution(k, tangerine):
    answer = 0
    num_of_tangerine = sorted(list(Counter(tangerine).values()), reverse=True)
    
    for num in num_of_tangerine:
        if k <= 0:
            break
        k -= num
        answer += 1
        
    return answer